source/blog/2019-05-11-rspec-rails-4-0-has-been-released.md

```diff
@@ -0,0 +1,181 @@
+---
+title: RSpec Rails 4.0 has been released!
+author: Benoit Tigeot, Jon Rowe and Phil Pirozhkov
+---
+
+RSpec Rails 4.0 has been released recently! Given our commitment to
+[semantic versioning](http://semver.org/), this update comes with 
+breaking changes for people that are using Rails below 4.2. If we 
+did introduce any bugs, please let us know, and we'll get 
+a patch release out with a fix ASAP.
+
+RSpec continues to be a community-driven project with contributors
+from all over the world. This release includes 258 commits and 56
+merged pull requests from 27 different contributors!
+
+Thank you to everyone who helped make this release happen!
+
+## Notable Changes
+
+Support for Rails 6, support for testing ActionCable, ActiveMailbox
+
+### Breaking changes
+
+RSpec Rails 4 supports only Rails 5 and 6. This is in line with our
+new versioning policy which keeps major versions of RSpec Rails in 
+lock step with versions of Rails.
+
+This means that we no longer support versions of Ruby below 2.2
+in line with the supported Ruby versions of Rails 5.
+
+We have not restricted the version of Rails/Ruby for this version to
+allow a "soft support" for Rails 4.2 during this transitionary
+period but we strongly urge you to upgrade Rails to use 4.0.0.
+
+### Improved JRuby support
+
+We are very happy to support JRuby on this new release. RSpec Rails 4
+supports JRuby with Rails 5.2 and 6.
+
+### Support for Action Cable
+
+Using a [channel spec][1] it will be easy to test your Action Cable
+channels. A `channels spec` is a thin wrapper for an
+`ActionCable::Channel::TestCase`, and includes all of the behavior
+and assertions that it provides, in addition to RSpec's own 
+behavior and expectations.
+
+It also includes helpers from `ActionCable::Connection::TestCase`
+to make it possible to test connection behavior.
+
+### Support for Action Mailbox
+
+Rails 6 came with [Action Mailbox][2]. We can now test them using
+`:mailbox` flag.
+
+```ruby
+```
+
+TODO: We do no have feature test for this...
+
+### Improvements on generators
+
+If you use generators, there is some improvements:
+- No more Ruby 1.9 hash syntax in file generated
+- Scaffold generator generates request specs instead of 
+controller spec
+- New [generators][3] available (channel, generator, mailbox...)
```

Comment:
```suggestion
If you use generators, there ds some improvements:
- The default hash syntax is now the Ruby 1.9 style in generated files to match Rails.
- Request specs are generated instead of controller specs by default
- New [generators][3] available (channel, generator, mailbox...)
```